src/main/kotlin/com/pubnub/api/crypto/CryptoModule.kt

```diff
@@ -131,6 +136,31 @@ class CryptoModule internal constructor(
     private fun getCryptorById(cryptorId: ByteArray): Cryptor? {
         return cryptorsForDecryptionOnly.firstOrNull { it.id().contentEquals(cryptorId) }
     }
+
+    private fun validateData(data: ByteArray) {
+        if (data.isEmpty()) {
+            throw PubNubException(
+                errorMessage = "Encryption/Decryption of empty data not allowed.",
+                pubnubError = PubNubError.ENCRYPTION_AND_DECRYPTION_OF_EMPTY_DATA_NOT_ALLOWED
+            )
+        }
+    }
+
+    private fun validateInputStreamAndReturnBuffered(stream: InputStream): BufferedInputStream {
+        // we use bufferedInputStream because we can read some data (from buffer) and pass them so then can be read again after reset()
+        val bufferedInputStream = stream.buffered()
+        bufferedInputStream.mark(Int.MAX_VALUE)
```

Comment:
If we know that only 1 byte will be read here then we don't need to use `Int.MAX_VALUE`. This might cost us some memory